packages/suite/src/hooks/wallet/coinmarket/form/common/useCoinmarketFormActions.ts

```diff
@@ -0,0 +1,221 @@
+import { selectAccounts, selectDevice } from '@suite-common/wallet-core';
+import {
+    amountToSatoshi,
+    formatAmount,
+    fromFiatCurrency,
+    isEthereumAccountSymbol,
+    isZero,
+} from '@suite-common/wallet-utils';
+import { BigNumber } from '@trezor/utils';
+import { FiatCurrencyCode } from 'invity-api';
+import { useCallback } from 'react';
+import { UseFormReturn } from 'react-hook-form';
+import {
+    FORM_CRYPTO_TOKEN,
+    FORM_OUTPUT_ADDRESS,
+    FORM_OUTPUT_AMOUNT,
+    FORM_OUTPUT_CURRENCY,
+    FORM_OUTPUT_FIAT,
+    FORM_OUTPUT_MAX,
+    FORM_SEND_CRYPTO_CURRENCY_SELECT,
+} from 'src/constants/wallet/coinmarket/form';
+import { useSelector } from 'src/hooks/suite';
+import { useCoinmarketFiatValues } from 'src/hooks/wallet/coinmarket/form/common/useCoinmarketFiatValues';
+import { useBitcoinAmountUnit } from 'src/hooks/wallet/useBitcoinAmountUnit';
+import { CoinmarketAccountOptionsGroupOptionProps } from 'src/types/coinmarket/coinmarket';
+import {
+    CoinmarketSellExchangeFormProps,
+    CoinmarketUseFormActionsProps,
+    CoinmarketUseFormActionsReturnProps,
+} from 'src/types/coinmarket/coinmarketForm';
+import { coinmarketGetSortedAccounts } from 'src/utils/wallet/coinmarket/coinmarketUtils';
+import {
+    cryptoToNetworkSymbol,
+    getNetworkDecimals,
+} from 'src/utils/wallet/coinmarket/cryptoSymbolUtils';
+
+/**
+ * shareable sub-hook used in useCoinmarketSellForm & useCoinmarketExchangeForm
+ * @return functions and values to handle form inputs and update fee levels
+ */
+export const useCoinmarketFormActions = <T extends CoinmarketSellExchangeFormProps>({
+    account,
+    methods,
+    setAmountLimits,
+    changeFeeLevel,
+    composeRequest,
+    setAccountOnChange,
+}: CoinmarketUseFormActionsProps<T>): CoinmarketUseFormActionsReturnProps => {
+    const { symbol } = account;
+    const { shouldSendInSats } = useBitcoinAmountUnit(symbol);
+    const accounts = useSelector(selectAccounts);
+    const device = useSelector(selectDevice);
+    const accountsSorted = coinmarketGetSortedAccounts({
+        accounts,
+        deviceState: device?.state,
+    });
+
+    const { getValues, setValue, clearErrors } =
+        methods as unknown as UseFormReturn<CoinmarketSellExchangeFormProps>;
+    const { outputs, sendCryptoSelect } = getValues();
+    const tokenAddress = outputs?.[0]?.token;
+    const tokenData = account.tokens?.find(t => t.contract === tokenAddress);
+    const isBalanceZero = tokenData
+        ? isZero(tokenData.balance || '0')
+        : isZero(account.formattedBalance);
+    const coinmarketFiatValues = useCoinmarketFiatValues({
+        accountBalance: account.formattedBalance,
+        cryptoSymbol: sendCryptoSelect?.value,
+        tokenAddress,
+        fiatCurrency: getValues().outputs?.[0]?.currency?.value as FiatCurrencyCode,
+    });
+    const networkDecimals = getNetworkDecimals(coinmarketFiatValues?.networkDecimals);
+
+    const onFiatCurrencyChange = useCallback(
```

Comment:
Consider _all_ of the occurrences for `useCallback` in this file. Are they really necessary?

It seems these functions are never listed as a dependency, or even passed as prop, always directly called at some place, so maybe there is no need to memoize? :thinking: